pkg/service/service.go

```diff
@@ -935,11 +943,17 @@ func (d *DynamicCRD) AddComponentLabelsToCRD(labels map[string]string) {
 
 // PushServiceFromKubernetesInlineComponents updates service(s) from Kubernetes Inlined component in a devfile by creating new ones or removing old ones
 // returns true if the component needs to be restarted (when a service binding has been created or deleted)
-func PushServiceFromKubernetesInlineComponents(client *kclient.Client, k8sComponents []devfile.Component, labels map[string]string) (bool, error) {
+func PushServiceFromKubernetesInlineComponents(client *kclient.Client, k8sComponents []devfile.Component, labels map[string]string, ownerReferences metav1.OwnerReference, deployment *v1.Deployment) (bool, error) {
```

Comment:
why this is suddenly accepting `deployment`?

Reply:
We use it to generate the owner reference and to regenerate the service binding object for deletion of it. If all the links are deleted, we can use this to get the deployment's name back.